lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -354,20 +356,30 @@ private function getShortName(string $className): string
         return strtolower(end($parts));
     }
 
+    /**
+     * Puts the `inherited` and `declared` values into mapping information for fields, associations
+     * and embedded classes.
+     *
+     * @param FieldMapping|AssociationMapping|mixed[] $mapping
```

Comment:
That union is basically the same as `mixed[]`. How does this help us?

Reply:
It was the only way I found that on the one hand satisfies CS requirements and on the other works for Psalm 🤷🏻‍♂️

Reply:
Maybe I missed what exactly the problem with PHPCS was. But unless we find a way to remove `mixed[]` from the union, it doesn't make sense to declare such a complex composite type.

Reply:
It works with `mixed[]` alone... is that ok?

Reply:
Sure, it's not worse than what we already have. 😅